kolibri/plugins/management/assets/src/views/user-page/delete-user-modal.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+
+  <core-modal :title="$tr('deleteUser')" @cancel="displayModal(false)">
+    <div>
+      {{ $tr('deleteConfirmation', { username }) }}
+      <div class="ta-r">
+        <k-button
+          :text="$tr('no')"
+          :primary="false"
+          :raised="false"
+          @click="displayModal(false)"
+        />
+        <k-button
+          :text="$tr('yes')"
+          :primary="true"
+          :raised="true"
+          :disabled="submitting"
+          @click="handleDeleteUser"
+        />
+      </div>
+    </div>
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  import { deleteUser, displayModal } from '../../state/actions';
+  import { kolibriLogout } from 'kolibri.coreVue.vuex.actions';
+  import coreModal from 'kolibri.coreVue.components.coreModal';
+  import kButton from 'kolibri.coreVue.components.kButton';
+
+  export default {
+    name: 'deleteUserModal',
+    components: {
+      coreModal,
+      kButton,
+    },
+    props: {
+      id: {
+        type: String,
+        required: true,
+      },
+      name: {
+        type: String,
+        required: true,
+      },
+      username: {
+        type: String,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        submitting: false,
+      };
+    },
+    methods: {
+      handleDeleteUser() {
+        this.submitting = true;
+        if (Number(this.id) === this.currentUserId) {
```

Comment:
As far as I know, `this.id` is a string guuid. Coercing it to Number would give you a NaN

Reply:
Fixed, moved logic to the deleteUser action.